### ISSUE TITLE
[ic-button] - accessibility error when setting `title` attribute on `icon` variant

### ISSUE THREAD

**Issue by ad9242** [mi6/ic-ui-kit#1845]
## Summary of the bug 
When an ic-button of `variant="icon"` has a `title` set, it causes as accessibility failure as there is no text visible to a screen reader.

## 🪜 How to reproduce 
Tell us the steps to reproduce the problem: 
1. Go to page in : ['storybook' ](https://mi6.github.io/ic-ui-kit/branches/develop/web-components/?path=/story/web-components-card--with-interaction-button)
2. Show the storybook add-ons if not visible
3. click on Accessibility tab
4. See the error 

also occurs with [icon buttons examples](https://mi6.github.io/ic-ui-kit/branches/develop/web-components/?path=/story/web-components-button--icon-only)

## 📸 Screenshots or code 
![image](https://github.com/mi6/ic-ui-kit/assets/116102637/3a71e7a3-5d5e-4f4a-a2e7-ce758da11bd8)

## 🧐 Expected behaviour 
Setting a `title` should not cause an accessibility error (setting an ariaLabel instead does not give an error)

## 📝 Acceptance Criteria  
If relevant, describe in full detail the different interactions and edge cases that the component or patterns needs to fulfil. 

*Given* an icon button exists
*When*  setting the `title` attribute to some text
*Then* the button should not have any accessibility failures

## Additional info 
This may have previously worked. Once fixed, a cypress test should be added - currently there are only tests for setting an ariaLabel on the icon buttons
